pkg/apis/core/validation/validation.go

```diff
@@ -1173,7 +1173,7 @@ func validateProjectedVolumeSource(projection *core.ProjectedVolumeSource, fldPa
 	return allErrs
 }
 
-var supportedHostPathTypes = sets.NewString(
+var supportedHostPathTypes = sets.New[string](
```

Comment:
This can be inferred

Edit: Why even convert this to string?  The natural type (`core.HostPathType`) is comparable, so satisfies the sets.Set type constraint.  L1190 would need to change to something like

```
func asStrings[T ~string](m map[T]bool) []string {
	var ret []string
	for k := range m {
		ret = append(ret, string(k))
	}
	return ret
}
```

and

```
field.NotSupported(fldPath, hostPathType, asStrings(sets.List(supportedHostPathTypes))
```

Reply:
Updated.